### PR TITLE
Introduce include prefix for python integration

### PR DESCRIPTION
**Pull request by jothepro**
We talked about [wether an include prefix for the support-lib should be configurable](https://github.com/cross-language-cpp/djinni-generator/pull/43#issuecomment-819169653). 

As I was pretty overwhelmed by your PR I thought it would give me a good insight into your changes if I implemented the discussed parameter. 

Now it's 3 hours later and I've introduced 4 parameters. 😆 

Added options:
* `--c-wrapper-header-out`
* `--c-wrapper-include-prefix`
* `--c-wrapper-include-cpp-prefix`
* `--c-wrapper-base-lib-include-prefix`

With these changes the cwrapper generator now supports similar configuration options as the other languages. The public header files of the c interface can be moved to a separate directory with `--c-wrapper-header-out`.
The cpp headers can also be in a different folder and included with `--c-wrapper-cpp-prefix`.
Baselib includes do now have the prefix `djinni/cwrapper`.

I also fixed the generator header, because I realized it wasn't grammatically correct. This makes the PR quite big as this affects every single generated file in `expected`. Sorry for that 🙈 ...

Furthermore I have fixed a few IntelliJ IDEA Code Analysis Warnings in the Python generator. I'm not good in Scala but I only applied recommendations that seemed to make sense to me.

### CODE REVIEW

File: src/it/resources/expected/all_datatypes/cpp-headers/all_datatypes.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // AUTOGENERATED FILE - DO NOT MODIFY!
-// This file generated by Djinni from all_datatypes.djinni
+// This file was generated by Djinni from all_datatypes.djinni
```
Comment (jothepro) [inline src/it/resources/expected/all_datatypes/cpp-headers/all_datatypes.hpp:2]: This is the change in the Djinni header that I did. It affects every single output file for the integration tests. I hope you agree that the sentence was not correct before this change. Sorry for bloating the PR with it, it's really just a small change that I had to fix once I saw it. 🙈

File: src/main/scala/djinni/Main.scala [hunk 2]
```diff
@@ -87,9 +87,12 @@ object Main {
     var yamlOutFile: Option[String] = None
     var yamlPrefix: String = ""
     var pyOutFolder: Option[File] = None
-    var pyPackageName: String = ""
```
Comment (jothepro) [inline src/main/scala/djinni/Main.scala:90]: `pyPackageName` is never used, so I removed it.

File: src/main/scala/djinni/Main.scala [hunk 3]
```diff
@@ -87,9 +87,12 @@ object Main {
...
     var pyIdentStyle = IdentStyle.pythonDefault
     var cWrapperOutFolder: Option[File] = None
+    var cWrapperHeaderOutFolderOptional: Option[File] = None
+    var cWrapperIncludePrefix: String = ""
+    var cWrapperIncludeCppPrefix: String = ""
+    var cWrapperBaseLibIncludePrefix: String = "djinni/cwrapper/"
```
Comment (jothepro) [inline src/main/scala/djinni/Main.scala:95]: The base-lib now has a default prefix of `djinni/cwrapper`. This should work with the existing support-lib.

File: src/main/scala/djinni/Main.scala [hunk 4]
```diff
@@ -108,13 +111,14 @@ object Main {
...
       head("djinni generator version", Main.getClass.getPackage.getImplementationVersion)
+      note("General")
       help("help")
       version("version")
       opt[File]("idl").valueName("<in-file>").required().foreach(idlFile = _)
         .text("The IDL file with the type definitions, typically with extension \".djinni\".")
       opt[String]("idl-include-path").valueName("<path> ...").optional().unbounded().foreach(x => idlIncludePaths = idlIncludePaths :+ x)
         .text("An include path to search for Djinni @import directives. Can specify multiple paths.")
-      note("")
+      note("\nJava")
```
Comment (jothepro) [inline src/main/scala/djinni/Main.scala:121]: I've added separators [like in the documentation](https://djinni.xlcpp.dev/djinni-generator/cli-usage/). Not related to your PR in any way but I think it makes the help more readable with it's ever growing number of configuration options.

File: src/main/scala/djinni/PythonGenerator.scala [hunk 5]
```diff
@@ -202,11 +202,10 @@ class PythonGenerator(spec: Spec) extends Generator(spec) {
...
 
     tm.base match {
-      case MList => {
-        writeChecksForContainer(tm.args(0), exclude, w)
-      }
+      case MList =>
+        writeChecksForContainer(tm.args.head, exclude, w)
       case MSet | MMap => {
-        writeChecksForContainer(tm.args(0), exclude, w)
+        writeChecksForContainer(tm.args.head, exclude, w)
```
Comment (jothepro) [inline src/main/scala/djinni/PythonGenerator.scala:208]: Just ignore these kind of changes, they are recommendations by IntelliJ IDEA. I guess this PR wasn't bloated enough already so I added them... 🤷

**Review by jothepro — COMMENTED** [review]
I have not had a chance to test if my changes produce valid output that can be compiled because I do not have a setup where I could test this yet. I am pretty confident about the changes but I'd be super happy if you could evaluate them for me on your machine... :/

**Comment by freitass** [comment]
Good catch on the hardcoded quotes! It works with the support lib PR.

**Pull request merged**